app/main.py

```diff
@@ -2,18 +2,13 @@
 
 import requests
 
+from app.exceptions import MissingAPIKeyError
+
 API_KEY = os.getenv("API_KEY")
 URL = "http://api.weatherapi.com/v1/current.json"
 CITY = "Paris"
 
 
-class MissingAPIKeyError(Exception):
-
-    def __init__(self, message: str) -> None:
-        super().__init__(message)
-        self.message = message
-
-
 def get_weather() -> None:
     if not API_KEY:
         raise MissingAPIKeyError("No API key provided for Weather API")
```

Comment:
Yaroslav meant that requests.get(URL, params=response_params(by the way, it should be request_params because you make a request)) may return 400 error and response will differ from what you expect. Like this
{
    "error": 'text'
}
or
{
     "errors": ['error 1', 'error2']
}
You don't know what kind of response you'll receive when get 400, 401, 500, etc errors, thus you should catch these errors. Like if status code isn't 200 -> print error. So check status code or catch error or use what Serhii recommended